### PR TITLE
Automate class hash computation in tests

### LINKED ISSUE #929: Automate class hash computation in tests
https://github.com/foundry-rs/starknet-foundry/pull/903#discussion_r1363382289

### PR DESCRIPTION
**Pull request by drknzz**
Closes #929 

## Introduced changes

-

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/cheatnet/tests/cheatcodes/declare.rs [hunk 1]
```diff
@@ -1,95 +1,86 @@
...
-            msg.to_string().contains("is already declared")
+            msg.to_string().contains(error_msg)
         }
         _ => false,
     });
 }
 
-#[test]
-fn declare_non_existant() {
+#[test_case("GoodbyeStarknet")]
```
Comment (Arcticae) [inline crates/cheatnet/tests/cheatcodes/declare.rs:71]: Why bother with that if it's not multiple contracts parametrized here?

File: crates/cheatnet/tests/cheatcodes/declare.rs [hunk 2]
```diff
@@ -1,95 +1,86 @@
...
+    contract_name: &str,
+    contracts: &HashMap<String, StarknetContractArtifacts>,
+) -> ClassHash {
+    let sierra = contracts.get(contract_name).unwrap();
+    get_class_hash(sierra.sierra.as_str()).unwrap()
+}
 
-#[test]
-fn declare_simple() {
+#[test_case("HelloStarknet")]
```
Comment (Arcticae) [inline crates/cheatnet/tests/cheatcodes/declare.rs:20]: Don't see a purpose in parametrizign that

File: crates/cheatnet/tests/cheatcodes/declare.rs [hunk 3]
```diff
@@ -1,95 +1,86 @@
...
-use starknet_api::hash::StarkFelt;
-use starknet_api::stark_felt;
-
-static HELLO_STARKNET_CLASS_HASH: &str =
-    "0x05ecd2c5f5ca68a4dc1b945582c69434adfc8bedbd188f0146a35875a4791936";
-static CONSTRUCTOR_SIMPLE_CLASS_HASH: &str =
-    "0x05f5abdb98a7ec93a20ac4b2ed284979613bc71f2dec3df10e589ebe40c562b4";
+use test_case::test_case;
+
+fn class_hash_from_str(
```
Comment (Arcticae) [inline crates/cheatnet/tests/cheatcodes/declare.rs:12]: This is not really from string

**Review by Arcticae — COMMENTED** [review]
Generally approve but some tidying up is needed

File: crates/cheatnet/tests/cheatcodes/declare.rs (hunk 1 above)
Comment (drknzz) [reply crates/cheatnet/tests/cheatcodes/declare.rs:71]: It's clearer to see the main constant arg at the top imo, should it be reverted?

File: crates/cheatnet/tests/cheatcodes/declare.rs (hunk 2 above)
Comment (drknzz) [reply crates/cheatnet/tests/cheatcodes/declare.rs:20]: It's the main context of the test; another contract can be easily tested without touching the test itself
Can revert it though

**Review by Arcticae — APPROVED** [review]

**Pull request merged**